From e5f68189ab3602b1042cc8423fd6a3a23abd5455 Mon Sep 17 00:00:00 2001
From: Alex James <theracermaster@gmail.com>
Date: Sat, 14 Apr 2018 22:59:57 -0500
Subject: [PATCH] ch341a_spi: Avoid deprecated libusb functions

libusb 1.0.22 marked libusb_set_debug as deprecated. For such versions
of libusb, use libusb_set_option instead.

Change-Id: Ib71ebe812316eaf49136979a942a946ef9e4d487
Signed-off-by: Alex James <theracermaster@gmail.com>
Reviewed-on: https://review.coreboot.org/25681
Tested-by: Nico Huber <nico.h@gmx.de>
Reviewed-by: David Hendricks <david.hendricks@gmail.com>
---
Upstream-Status: Backport
 ch341a_spi.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/ch341a_spi.c b/ch341a_spi.c
index 6eb2804..d880510 100644
--- a/ch341a_spi.c
+++ b/ch341a_spi.c
@@ -444,7 +444,12 @@ int ch341a_spi_init(void)
 		return -1;
 	}
 
-	libusb_set_debug(NULL, 3); // Enable information, warning and error messages (only).
+	/* Enable information, warning, and error messages (only). */
+#if LIBUSB_API_VERSION < 0x01000106
+	libusb_set_debug(NULL, 3);
+#else
+	libusb_set_option(NULL, LIBUSB_OPTION_LOG_LEVEL, LIBUSB_LOG_LEVEL_INFO);
+#endif
 
 	uint16_t vid = devs_ch341a_spi[0].vendor_id;
 	uint16_t pid = devs_ch341a_spi[0].device_id;
-- 
2.17.0

